src/g3log/logworker.hpp

```diff
@@ -109,6 +109,13 @@ namespace g3 {
          removeWrappedSink(sink_handle->sink());
       }
 
+      void flush(bool propogate_to_sinks = true) {
+         g3::spawn_task([propogate_to_sinks] {
+            if (propogate_to_sinks) {
+               // TODO: ...

```

Comment:
I prefer explicit APIs to default parameters. 

If you want something to is passed onto the sinks to ensure they are synched also it should instead be split into two functions following the SRP principle. 

pseudo code
```
void flushWorker() {
   auto doNothing = []() {};
   g3::spawn_task(doNothing).wait();
```

```
void flushAllSinks() {
  auto synchronized = []() {
     for (auto& sink : _impl._sinks) {
        sink->flush();
     }
};

spawn_task(synchronized).wait();
}
```